### ISSUE TITLE
KubernetesServiceImporter: Record is not unpublished if removed as result of MODIFIED event

### ISSUE THREAD

**Issue by daka1510** [vert-x3/vertx-service-discovery#96]
**Reproduction:**
- deploy a service (e.g. hello-world) that uses the `KubernetesServiceImporter`
- deploy another service (e.g. hello-minikube) in the same namespace
- delete the second service and observe the received updates (`KubernetesServiceImporter#onChunk`):
```
MODIFIED Event:
{
  "location": {
    "endpoint": "http://10.98.7.139:8080",
    "host": "10.98.7.139",
    "port": 8080,
    "root": "",
    "ssl": false
  },
  "metadata": {
    "run": "hello-minikube",
    "kubernetes.namespace": "default",
    "kubernetes.name": "hello-minikube",
    "kubernetes.uuid": "37c57c1e-deb0-11e8-a8ee-0800274f8294"
  },
  "name": "hello-minikube",
  "status": "UNKNOWN",
  "type": "http-endpoint"
}


DELETED Event:
{
  "location": {
    "endpoint": "http://10.98.7.139:8080",
    "host": "10.98.7.139",
    "port": 8080,
    "root": "",
    "ssl": false
  },
  "metadata": {
    "run": "hello-minikube",
    "kubernetes.namespace": "default",
    "kubernetes.name": "hello-minikube",
    "kubernetes.uuid": "37c57c1e-deb0-11e8-a8ee-0800274f8294"
  },
  "name": "hello-minikube",
  "status": "UNKNOWN",
  "type": "http-endpoint"
}
```

**Issue**
- The Record (here: http://10.98.7.139:8080 // 37c57c1e-deb0-11e8-a8ee-0800274f8294) is deleted but `KubernetesServiceImporter.java#unpublishRecord` is never invoked. Consequence: the service is still available via service discovery APIs.

**Root Cause**
- as result of the `MODIFIED` event the storedRecord gets removed (https://github.com/vert-x3/vertx-service-discovery/blob/master/vertx-service-discovery-bridge-kubernetes/src/main/java/io/vertx/servicediscovery/kubernetes/KubernetesServiceImporter.java#L271) 
- when the `DELETE` event is processed shortly after, no storedRecord is found and therefore https://github.com/vert-x3/vertx-service-discovery/blob/master/vertx-service-discovery-bridge-kubernetes/src/main/java/io/vertx/servicediscovery/kubernetes/KubernetesServiceImporter.java#L265 is never called

**Comment by daka1510** [comment]
@cescoffier - took me some time to find out what's happening but now I'msure it's a bug in the code here. Is the description detailed enough to follow along?

**Comment by daka1510** [comment]
@cescoffier - I don't think it's an issue to test for UUID and endpoint.

The modified and deleted events above were sent by the Kubernetes API when a single service was deleted. That's why name and endpoint match.

**Comment by cescoffier** [comment]
Oh, I see, so when deleting the service, the 2 events are sent. 

the bug is `storedRecord = removeRecordIfContained(record);` - under modification, it must not remove it, but updated it and re-publish it.

**Comment by cescoffier** [comment]
PR: https://github.com/vert-x3/vertx-service-discovery/pull/97.

**Comment by cescoffier** [comment]
Fixed by #97
